realm/realm-library/src/main/cpp/io_realm_internal_objectstore_OsRemoteMongoCollection.cpp

```diff
@@ -33,13 +33,36 @@
 
 using namespace realm;
 using namespace realm::app;
+using namespace realm::bson;
 using namespace realm::jni_util;
 using namespace realm::_impl;
 
-static std::function<jobject(JNIEnv*, uint64_t)> collection_mapper = [](JNIEnv* env, uint64_t result) {
+// This mapper works for both count and delete operations
+static std::function<jobject(JNIEnv*, uint64_t)> collection_mapper_count = [](JNIEnv* env, uint64_t result) {
     return JavaClassGlobalDef::new_long(env, result);
 };
 
+static std::function<jobject(JNIEnv*, util::Optional<bson::BsonDocument>)> collection_mapper_find_one = [](JNIEnv* env, util::Optional<bson::BsonDocument> document) {
+    return document ? JniBsonProtocol::bson_to_jstring(env, *document) : NULL;
+};
+
+static std::function<jobject(JNIEnv*, util::Optional<ObjectId>)> collection_mapper_insert_one = [](JNIEnv* env, util::Optional<ObjectId> object_id) {
+    return JavaClassGlobalDef::new_object_id(env, object_id.value());
```

Comment:
Though technically possible, I doubt we can ever reach it. I'm guessing the only way `object_id` has no value is because there has been an error inserting the Bson, in which case the error callback will be triggered, e.g. inserting the same document with the same ID twice. I will make it throw an exception from here just in case:
```
    if (object_id) {
        return JavaClassGlobalDef::new_object_id(env, object_id.value());
    }
    throw util::invalid_argument("Error in 'insert_one', parameter 'object_id' has no value.");
```

Reply:
Fixed using suggestion from the comment below.